src/scripts/main.js

```diff
@@ -1,3 +1,28 @@
 'use strict';
 
-// write code here
+const listEl = document.querySelector('ul');
+
+const convertToNumber = (str) => Number(str.replace('$', '').replace(',', ''));
+
+const sortList = (list, sortParam) =>
+  list.sort(
+    (a, b) =>
+      convertToNumber(b.dataset[sortParam]) - convertToNumber(a.dataset[sortParam]),
+  );
+
+const getEmployees = (sortedList, element) => (element.innerHTML = sortedList);
+
+const visibleEmployeesHTML = sortList([...listEl.children], 'salary')
+  .map(
+    (item) =>
+      `<li
+        data-position="${item.dataset.position}"
+        data-salary="${item.dataset.salary}"
+        data-age="${item.dataset.age}"
+      >
+        ${item.innerText}
+      </li>`,
+  )
+  .join('');
```

Comment:
Here you are constructing a string of HTML to represent the sorted list of employees. This approach is generally not recommended because it's prone to XSS attacks and it's harder to manage compared to working with DOM elements. Consider creating and appending DOM elements instead of constructing a string of HTML.